include/tarxx.h

```diff
@@ -199,16 +242,41 @@ namespace tarxx {
         static constexpr unsigned int HEADER_LEN_SIZE = 11U;
         static constexpr unsigned int HEADER_LEN_MTIM = 11U;
 
-        void write(const block_t& data)
+        void lz4_flush()
+        {
+            lz4_out_buf_pos_ += lz4_check_error(LZ4F_flush, lz4_ctx_->get(), lz4_out_buf_.data(), lz4_out_buf_.capacity(), nullptr);
+            const auto lz4_result = lz4_check_error(LZ4F_flush, lz4_ctx_->get(), lz4_out_buf_.data(), lz4_out_buf_.capacity(), nullptr);
+            lz4_out_buf_pos_ += lz4_result;
+            write_lz4_data();
+        }
+
+        void write(const block_t& data, bool is_header = false)
         {
             if (!is_open()) return;
-            switch (mode_) {
-                case output_mode::stream_output:
-                    callback_(data);
-                    break;
-                case output_mode::file_output:
-                    file_.write(data.data(), data.size());
-                    break;
+            if (compression_ == compression_mode::lz4) {
+                if (is_header) {
+                    const auto lz4_result = lz4_check_error(LZ4F_uncompressedUpdate, lz4_ctx_->get(), lz4_out_buf_.data(), lz4_out_buf_.capacity(),
+                                                            data.data(), data.size(), nullptr);
+
+                    lz4_out_buf_pos_ += lz4_result;
+                    // flush is necessary to keep lz4_out_buf_pos_ consistent
+                    lz4_flush();
+                } else {
+                    const auto lz4_result = lz4_check_error(LZ4F_compressUpdate, lz4_ctx_->get(), lz4_out_buf_.data(), lz4_out_buf_.capacity(),
+                                                            data.data(), data.size(), nullptr);
+                    lz4_out_buf_pos_ += lz4_result;
+                }
+                write_lz4_data();
+            } else {
+                switch (mode_) {
+                    case output_mode::stream_output:
+                        callback_(data, data.size());
+                        break;
+                    case output_mode::file_output:
+                        file_.write(data.data(), data.size());
+                        file_.flush(); // todo remove this
```

Comment:
do we really need this? maybe follow the todo? :-)

Reply:
removed flush and lz4 compression still works :+1: